Theory_assignment_two/assignment_2.md

```diff
@@ -0,0 +1,453 @@
+| Program is subdivided into objects                                       | Program is subdivided into objects                        |   |
+| It is less secure because it doesn’t have a way of hiding data           | It is more secure because it has a way of hiding data     |   |
+| Doesn’t have concept of access specifiers                                | Uses access specifiers for encapsulation                  |   |
+| Languages that use procedure-oriented programming: FORTRAN, ALGOL, COBOL | Languages that use OOP programming: Python, Java, C++, C# |   |
+| Overloading is not possible                                              | Overloading is possible                                   |   |
+| No dating hiding and inheritance                                         | Has data hiding and inheritance                           |   |
+| Uses the concept of procedure abstraction                                | Uses concept of data abstraction                          |   |
+| Adding new data and functions is not easy                                | Adding new data and functions is easy                     |   |
+
+
+## 2. What's polymorphism in OOP? 
+
+Polymorphism is the condition of occurrence in different forms. It is the use of a single type entity (method, operator or object) to represent different scenarios.
+
+Example of an inbuilt function with polymorphism is the len() function.
+```python
+# #inbuilt function len() with polymorphism
+#
+len("hello")
+len(["Dog","Cat","Mouse"])
+
+```
+
+Polymorphism is implemented through method overloading and method overriding.
+
+Method overriding is a method that uses the same name in both the parent and child class. 
+
+Example:
+```python
+# class Animal:
+#     def print_details(self):
+#         print("This is parent Animal class method")
+#
+#
+# class Horse(Animal):
+#     def print_details(self):
+#         print("This is child Horse class method")
+#
+#
+# class Cat(Animal):
+#     def print_details(self):
+#         print("This is child Cat class method")
+#
+#
+# animal_a = Animal()
+# animal_a. print_details()
+#
+# horse_b = Horse()
+# horse_b.print_details()
+#
+# cat_c = Cat()
+# cat_c.print_details()
+#
+```
+Method overloading is a property of a method to behave differently depending on number and types of parameters.
+
+Example:
+
+```python
+ # method overloading
+#
+# class Student:
+#     def __init__(self, m1, m2):
+#         self.m1 = m1
+#         self.m2 = m2
+#
+#     def sum(self, a=None, b=None, c=None):
+#         s = 0
+#         if a!=None and b!=None and c!=None:
+#             s = a + b + c
+#         elif a!=None and b!=None:
+#             s = a + b
+#         else:
+#             s = a
+#         return s
+#
+# studentsum_2 = Student(50,12)
+#
+# print(studentsum_2.sum(4,6,5))
+#
+
+```
+
+## 3. What's inheritance in OOP? 
+
+It is the transfer of the characteristics of a class to other classes that are derived from it.
```

Comment:
8/10 Examples are great. However, missing more details on what is the definition of inheritance. Could explain more on what can be inherited to the child classes i.e. attributes and behaviours.